datafusion/physical-plan/src/aggregates/group_values/single_group_by/primitive/mod.rs

```diff
@@ -130,15 +133,15 @@ where
                     let hash = key.hash(state);
                     let insert = self.map.entry(
                         hash,
-                        |&(g, _)| unsafe { self.values.get_unchecked(g).is_eq(key) },
```

Comment:
I wonder if the existing performance couldn't already be improved by doing `h == hash && self.values.get_unchecked(g).is_eq(key)` (i.e. avoiding the fetch if hash equality doesn't hold), like we do in other places. This already filters out most of them as hashes are almost never the same.